rust/lance/src/io/exec/scan.rs

```diff
@@ -367,7 +359,38 @@ impl LanceStream {
                 // We must be waiting to finish a file before moving onto thenext. That's an issue.
                 .try_flatten()
                 // We buffer up to `batch_readahead` batches across all streams.
-                .try_buffered_with_ordering(fragment_readahead, config.ordered_output)
+                .try_buffered(config.batch_readahead)
```

Comment:
I don't know if this was the problem (I assume the user isn't using v1 files?) but it does look like a `fragment_readahead` got used instead of a `batch_readahead` when you made the original change.